src/db/util/index.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Take a collection of objects,
+ * and create a map of them, using aparticular property as the key
+ */
+export const organizeObjectsByUniqueProperty = <I, P extends keyof I>(
```

Comment:
this function can apply more generally than jsut `db` related actions, so should probably live in a module in `src/util` instead.